src/v/kafka/client/client.cc

```diff
@@ -147,6 +147,10 @@ ss::future<> client::mitigate_error(std::exception_ptr ex) {
           try {
               std::rethrow_exception(ex);
           } catch (const broker_error& ex) {
+              if (ex.error == error_code::network_exception) {
+                  vlog(kclog.warn, "broker_error: {}", ex);
+                  return ss::make_exception_future(ex);
+              }
```

Comment:
I may have to have another think about this. The issue may have been introduced in 3ca886134ec3f5e0df4b2b53f097c5d68237b4c3

Reply:
For posterity, the fix is here: https://github.com/redpanda-data/redpanda/pull/7735